engine/runtime-instrument-id-execution/src/main/java/org/enso/interpreter/instrument/IdExecutionInstrument.java

```diff
@@ -75,13 +75,10 @@ private static class IdEventNodeFactory implements ExecutionEventNodeFactory {
      * @param callbacks communication with users
      * @param timer the timer for timing execution
      */
-    IdEventNodeFactory(
-      CallTarget entryCallTarget,
-      Callbacks callbacks,
-      Timer timer
-    ) {
+    IdEventNodeFactory(CallTarget entryCallTarget, Callbacks callbacks, ExecutionSupport executionSupport, Timer timer) {
```

Comment:
Delete `ExecutionSupport` and pass here one additional argument:
- `String inlineExpression` - the expression to evalute

Reply:
It's more complicated than that. While I can pass the `VisualisationHolder` with the expression, I also need a callback that will handle the result of the evaluation.